raml/schemas/definitions/analysis.json

```diff
@@ -0,0 +1,83 @@
+{
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "type": "object",
+    "definitions": {
+		"inputs": 		{
+							"type": ["array", "null"],
+						 	"items": {
+						 		"$ref": "file.json"
+							}
+						},
+		"outputs": 		{
+							"type": ["array", "null"],
+						 	"items": {
+						 		"$ref": "file.json"
+							}
+						},
+		"notes":        {"type": "array",  "items": {"$ref": "note.json"}},
+		"description" : {"type": "string"},
+        "job":			{"type": "string"},
+        "label":    	{"type": "string", "minLength": 1, "maxLength": 256},
+        "user":     	{"type": "string"},
+		"analysis-input":{
+			"type":"object",
+			"properties":{
+				"inputs":{"$ref":"#/definitions/inputs"},
+				"outputs":{"$ref":"#/definitions/outputs"},
+				"notes":{"$ref":"#/definitions/notes"},
+				"description":{"$ref":"#/definitions/description"},
+				"job":{"$ref":"#/definitions/job"},
+				"label":{"$ref":"#/definitions/label"},
+				"user":{"$ref":"#/definitions/user"}
+			},
+		    "additionalProperties": false
+		},
+		"analysis-output":{
+			"type":"object",
+			"properties":{
+				"_id":{"$ref":"../definitions/objectid.json#"},
+				"files":{
+					"type":"array",
```

Comment:
No because file.json file-output definition contains additionalProperties: false, we have to re-list all the properties.  I did $ref each individual property definition from file.json which is analogous to using allOf